kie-server-parent/kie-server-tests/kie-server-integ-tests-drools/pom.xml

```diff
@@ -152,6 +152,7 @@
         <configuration>
           <container>
             <systemProperties>
+              <org.kie.server.strict.javaBeans.serializers>true</org.kie.server.strict.javaBeans.serializers>
```

Comment:
If I'm not mistaken, this changes ALL the kie-server Drools integration test to use the new configuration flag which was needed for JBPM-9784.

I am not sure this is desirable, that flag is for jBPM/RHPAM and it's best if only pertaining to the original scope of specific customer request.

I would keep the Drools Integration Test as we provided with
 - https://github.com/kiegroup/droolsjbpm-integration/pull/2595

and without the need to change the configuration behaviour of Drools, DMN, etc. kie-server integratio tests.

my2c
/cc @tkobayas @danielezonca @mariofusco @lucamolteni

Reply:
as an integration test, we cannot have both behaviors based on a property in the same module. Perhaps it's enough to test it at marshaller, removing these integration tests if you don't want to alter former behavior

Reply:
the Kie Server Drools IntegrationTest we introduced on Drools side with 
 - https://github.com/kiegroup/droolsjbpm-integration/pull/2595

are needed because:
 - we did not have a test with a Business / Domain POJO with `CapitalizedField`
 - we did not have a test for ensuring non-regression of the JSON payload, as detailed in the original JIRA https://issues.redhat.com/browse/DROOLS-6592

I still am under the impression this flag would not be needed for Drools kie server IT tests (imho)

Reply:
@tarilabs it is not relevant whether that flag is active or not as long as we have tests covering both cases (which is the case). 
The flag is about kie server not just jBPM so it covers every aspect of kiesever not just the engine.

Reply:
Let me try to bring a specific concern, with this highlighted line.

We always ask SA, Fields, and Users, to log-dump the Kie Server Client Java API request/response first, if they want to directly interact with a kie-server without using our java Client. ie: log-dump the request/response, and modify as needed from a base, not build a request payload from scratch.

If this line is merged, then people using these ITs in the directory
```
kie-server-parent/kie-server-tests/kie-server-integ-tests-drools
```
which is about Drools kie-server IT tests,
those using these ITs for their reference would start to use a payload shape which is NOT the default (I am assuming most of Users would NOT set the `org.kie.server.strict.javaBeans.serializers` overriding the default)

Hope this clarifies the concern, and the reasons of the original IT tests inside 
- https://github.com/kiegroup/droolsjbpm-integration/pull/2595

Thank you for considering this aspect too

Reply:
removing your reproducer then as JSON Marshaller is already having those tests in place.